### PR TITLE
Add storage connector (https://github.com/snowlift/trino-storage) to Trino image

### PR DESCRIPTION
**Pull request by soenkeliebau**
This requires the storage connector to be uploaded at `https://repo.stackable.tech/repository/packages/trino-storage/trino-storage-${VERSION}.zip` for all supported Trino versions.

### CODE REVIEW

File: trino/Dockerfile [hunk 1]
```diff
@@ -38,6 +38,9 @@ RUN curl https://repo.stackable.tech/repository/packages/jmx-exporter/jmx_promet
 
 RUN curl -L https://repo.stackable.tech/repository/packages/trino-opa-authorizer/trino-opa-authorizer-${PRODUCT}-${OPA_AUTHORIZER}.tar.gz | tar -xzC /stackable/trino-server/plugin
 
+RUN curl https://repo.stackable.tech/repository/packages/trino-storage/trino-storage-${PRODUCT}.zip -o /tmp/trino-storage-${PRODUCT}.zip && \
```
Comment (sbernauer) [inline trino/Dockerfile:41]: We should remove the .zip afterwards

File: trino/CHANGELOG.md [hunk 2]
```diff
@@ -7,8 +7,10 @@ All notable changes to this project will be documented in this file.
 ### Added
 
 - Add `htpasswd` tool ([#385]).
+- Experimental: Add Snowlift Trino Storage Connector (https://github.com/snowlift/trino-storage) ([#397])
```
Comment (sbernauer) [inline trino/CHANGELOG.md:10]:
```suggestion
- Experimental: Add [Snowlift Trino Storage Connector](https://github.com/snowlift/trino-storage) ([#397])
```

**Review by sbernauer — CHANGES_REQUESTED** [review]

File: trino/Dockerfile (hunk 1 above)
Comment (soenkeliebau) [reply trino/Dockerfile:41]: done

File: trino/CHANGELOG.md [hunk 3]
```diff
@@ -7,8 +7,10 @@ All notable changes to this project will be documented in this file.
 ### Added
 
 - Add `htpasswd` tool ([#385]).
+- [EXPERIMENTAL] Add [Snowlift Trino Storage Connector](https://github.com/snowlift/trino-storage) ([#397])
```
Comment (sbernauer) [inline trino/CHANGELOG.md:10]:
```suggestion
- [EXPERIMENTAL] Add [Snowlift Trino Storage Connector](https://github.com/snowlift/trino-storage), but only for Trino version 414 ([#397])
```

**Review by sbernauer — DISMISSED** [review]

**Review by sbernauer — APPROVED** [review]

**Pull request merged**